tools/go-cli/Dockerfile

```diff
@@ -34,33 +34,70 @@ RUN mkdir /src/github.com/go-whisk-cli/build
 
 # The BUILDTIME argument is used as the CLI build time property (wsk property get --all)
 # Disabled for now to speed build times
-#ARG BUILDTIME
+ARG LOCAL_BUILD
+ARG LOCAL_OS
+ARG LOCAL_ARCH
+
+RUN echo $LOCAL_ARCH
 
 # Build the Go wsk CLI binaries
-ENV GOOS=windows
 ENV EXEC=wsk.exe
-ENV ARCHS="386 amd64"
-RUN for GOARCH in $ARCHS; do \
+RUN if [ "$LOCAL_BUILD" = "true" ]; then \
+      if [ "$LOCAL_OS" = "windows" ] ; then \
+        export GOOS=$LOCAL_OS && \
+        export ARCHS=$LOCAL_ARCH; \
+      else \
+        unset GOOS && \
+        unset ARCHS; \
+      fi; \
+    else \
+      export GOOS=windows && \
+      export ARCHS="386 amd64"; \
+    fi && \
```

Comment:
A lot of duplication in these, can we have only on if disambiguating this?

```
if [ "$LOCAL_BUILD" = "true" ]; then \
  export OSS=$LOCAL_OS && \
  export ARCHS=$LOCAL_ARCH; \
else \
  export OS="windows darwin linux" && \
  export ARCHS="386 amd64"; \
fi && \


for GOOS in $OSS; do \
  EXEC=wsk \
  if [ "$GOOS" = "windows" ]; then \
    EXEC="${EXEC}.exe" \
  fi \
  for GOARCH in $ARCHS; do \
    cd /src/github.com/go-whisk-cli && \
    go build -ldflags "-X main.CLI_BUILD_TIME=`date -u '+%Y-%m-%dT%H:%M:%S%:z'`" -v -o build/$GOOS/$GOARCH/$EXEC main.go && \
    if [ "$GOOS" = "windows"]; then \
      zip build/$GOOS/$GOARCH/openwhisk-win-$GOARCH.zip build/$GOOS/$GOARCH/$EXEC; \
    else \
      if [ "$GOOS" = "darwin" ]; then; export GOOS=mac; fi \
      tar -cf build/$GOOS/$GOARCH/openwhisk-$GOOS-$GOARCH.tar build/$GOOS/$GOARCH/$EXEC && \
      gzip < build/$GOOS/$GOARCH/openwhisk-$GOOS-$GOARCH.tar > build/$GOOS/$GOARCH/openwhisk-$GOOS-$GOARCH.tar.gz && \
      rm -f build/$GOOS/$GOARCH/openwhisk-$GOOS-$GOARCH.tar; \
    fi \
  done \
done \
```

should be roughly the same as the three different `RUN` statements. Will safe layers and is easier to maintain imho. (Not tested, highly likely there are errors in this)